telemetry/client.go

```diff
@@ -56,3 +57,44 @@ func (c *Client) PushReceivedMessages(filter transport.Filter, sshMessage *types
 		c.logger.Error("Error sending message to telemetry server", zap.Error(err))
 	}
 }
+
+func (c *Client) PushReceivedEnvelope(envelope *protocol.Envelope) {
+	c.logger.Debug("Pushing received envelope to telemetry server")
+	url := fmt.Sprintf("%s/received-envelope", c.serverURL)
+	postBody := map[string]interface{}{
+		"messageHash":    types.EncodeHex(envelope.Hash()),
```

Comment:
should this be `envelopeHash` or you're reusing a similar API to the message one?

Reply:
Hmm, I called it messageHash also in database. Technically envelope's hash IS message's hash.
So I am not sure, if you think that envelopeHash would be better, then I can change it.